### PR TITLE
Mobile code validation fix

### PR DESCRIPTION
**Pull request by sowmiyasubramaniyan**

### CODE REVIEW

**Review by SarfrazS — CHANGES_REQUESTED** [review]
Still opt verification popup page, validation not working it is accepting +91 see in below image.
image
image

**Comment by SarfrazS** [comment]
Screenshot 2024-01-23 185925
Screenshot 2024-01-23 185938
Screenshot 2024-01-23 185954

**Review by SarfrazS — CHANGES_REQUESTED** [review]
Pointed issues are fixed now on below places.
1. Registration page.
2. Login page
3. Verification page.
4. Contact us page.
5. Request demo page.
6. Create and edit contact page.
image

**Review by SarfrazS — DISMISSED** [review]
1st and 2nd Round Test summary:
Total testing time in days: 3
Total issues : 3
No of times PR went for functionality correction: 2
PR breaks existing functionality: No

File: app/Http/Controllers/Auth/AuthController.php [hunk 1]
```diff
@@ -126,7 +127,9 @@ public function requestOtp(Request $request)
             'mobile' => 'required|numeric',
         ]);
         $newNumber = ltrim($request->newnumber, '0');
-        User::where('id', $request->id)->update(['mobile' => $newNumber]);
+        $newCode = ltrim($request->code, '0');
```
Comment (sandesh556) [inline app/Http/Controllers/Auth/AuthController.php:130]: This code is repeating, you can create a protected method in parent class and call it in both parent and child class

**Review by sandesh556 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Auth/AuthController.php (hunk 1 above)
Comment (sowmiyasubramaniyan) [reply app/Http/Controllers/Auth/AuthController.php]: No need of ltrim() function for mobile, so removed, just mentioned as $request->code.

Comment (sandesh556) [reply app/Http/Controllers/Auth/AuthController.php]: Not ltrim, i am talking about the whole code block

File: app/Http/Controllers/Auth/AuthController.php [hunk 2]
```diff
@@ -126,7 +127,9 @@ public function requestOtp(Request $request)
             'mobile' => 'required|numeric',
         ]);
         $newNumber = ltrim($request->newnumber, '0');
-        User::where('id', $request->id)->update(['mobile' => $newNumber]);
+        $newCode = $request->code;
```
Comment (sandesh556) [inline app/Http/Controllers/Auth/AuthController.php:130]: This code block

Comment (sandesh556) [inline app/Http/Controllers/Auth/AuthController.php:130]: This code block

**Review by sandesh556 — CHANGES_REQUESTED** [review]

**Pull request merged**